### PR TITLE
Complete purchase flow before confirmation

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Reset the current shop session at the end of the checkout.

Update confirmation page URL to include Shop Session ID.

## Justify why they are needed

After checkout is complete we need to setup a new Shop Session (using the auth token). This will be done as soon as we navigate to the next page (confirmation).

The confirmation page needs to identify the relevant shop session by the ID in the URL instead of from cookies. This also makes the confirmation page behave more like a recipt that you can get back to.

TODO: make confirmation page a protected route when we can link a Shop Session to a specific member.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

[GRW-1577]: https://hedvig.atlassian.net/browse/GRW-1577?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #610** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/610" target="_blank">Graphite</a> 
    * **PR #613** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/613" target="_blank">Graphite</a> 
      * **PR #620** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/620" target="_blank">Graphite</a> 
        * **PR #622** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/622" target="_blank">Graphite</a> 
          * **PR #629** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/629" target="_blank">Graphite</a> 
            * **PR #645** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/645" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/645?utm_source=stack-comment).

File: apps/store/src/pages/confirmation/[shopSessionId].tsx [hunk 1]
```diff
@@ -6,21 +6,25 @@ import { ConfirmationPageProps } from '@/components/ConfirmationPage/Confirmatio
...
 
-  const apolloClient = initializeApollo(undefined, req, res)
+  const shopSessionId = params?.shopSessionId
+  if (!shopSessionId) return { notFound: true }
 
+  const apolloClient = initializeApollo(undefined, req, res)
+  const shopSessionService = setupShopSessionServiceServerSide({ apolloClient, req, res })
   const [shopSession, globalStory] = await Promise.all([
-    getCurrentShopSessionServerSide({ req, res, apolloClient }),
+    shopSessionService.fetchById(shopSessionId),
```
Comment (alebedev) [inline apps/store/src/pages/confirmation/[shopSessionId].tsx:27]: Let's handle the case when backend return 404 for our shop session

**Review by alebedev — APPROVED** [review]
Looks good

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/645?utm_source=stack-comment).


Job Reference: otDkpy8tB4rmLMp8y6We

**Pull request merged**